File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    # n마리 중 n/2 마리 가져가는 것이 최대값
    max_ans = len(nums)//2
    
    # 폰켓몬 개수 저장
    d = {}
    
    for n in nums:
        if n not in d:
            d[n]=1
        else:
            d[n]+=1
    if len(d) >= max_ans:
        return max_ans
    else:
        return len(d)
        

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        // 1. 중복 제거한 포켓몬 종류 set 만들기
        HashSet<Integer> set = new HashSet<>();
        
        for(int num: nums){
            set.add(num);            
        }
        System.out.println(set);
        // 2. 고르려는 마리수
        int select = nums.length/2;
    
        // 3. 최대 고르는 마릿수 return
        if(select <= set.size())
            return select;
        else
            return set.size();
    }
}